data/data/bootstrap/files/usr/local/bin/bootkube.sh.template

```diff
@@ -368,16 +370,36 @@ fi
 
 echo "Starting cluster-bootstrap..."
 
-if [ ! -f cb-bootstrap.done ]
-then
-    bootkube_podman_run \
+run_cluster_bootstrap() {
+	bootkube_podman_run \
         --rm \
         --volume "$PWD:/assets:z" \
         --volume /etc/kubernetes:/etc/kubernetes:z \
         "${CLUSTER_BOOTSTRAP_IMAGE}" \
         start --tear-down-early=false --asset-dir=/assets --required-pods="${REQUIRED_PODS}"
+}
 
-    touch cb-bootstrap.done
+gather_bootstrap_on_failure() {
+	cluster_bootstrap_gather
+	if (( $(stat -c%s '/tmp/artifacts-api-server-temp/resources/openapi.json.gz') <= 20 ))
```

Comment:
Move this logic into `cluster_bootstrap_gather`. The same logic should happen regardless of whether `cluster_bootstrap_gather` is run after a failed cluster bootstrap or as part of responding to a bootstrap gather request.